### PR TITLE
[pinmux/doc] Add pinmux testplan

### PR DESCRIPTION
**Pull request by cindychip**
This PR updates pinmux testplan.

Signed-off-by: Cindy Chen <chencindy@opentitan.org>

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM

**Review by msfschaffner — APPROVED** [review]
LGTM, thanks!

File: hw/ip/pinmux/data/pinmux_fpv_testplan.hjson [hunk 1]
```diff
@@ -5,121 +5,795 @@
...
+      tests: ["pinmux_assert"]
+    }
+
+    // `mio_out_o` in sleep mode tests
+    // Symbolic variable `mio_sel_i` is used to select a specific `mio_out_o` pin.
+    {
+      name: MioSleepMode0_A
+      desc: '''At posedge of `sleep_en_i`, if register `mio_pad_sleep_en` is 1 and
+            `mio_pad_sleep_mode` is 0, which means the pad is driven zero in deep sleep mode.
+            In the meantime, if register `mio_pad_sleep_status` is not written via TLUL interface
```
Comment (matutem) [inline hw/ip/pinmux/data/pinmux_fpv_testplan.hjson:157]:
```suggestion
            If, in the meantime, register `mio_pad_sleep_status` is not written via TLUL interface
```
And below...

File: hw/ip/pinmux/data/pinmux_fpv_testplan.hjson [hunk 2]
```diff
@@ -5,121 +5,795 @@
...
+            mode.
+            In the meantime, if register `mio_pad_sleep_status` is not written via TLUL interface
+            to clear the sleep status, the corresponding `mio_out_o` should be stable.
+            '''
+      milestone: V1
+      tests: ["pinmux_assert"]
+    }
+    {
+      name: MioSleepStable_A
+      desc: '''If not at posedge of `sleep_en_i`, and in the meantime, if register
```
Comment (matutem) [inline hw/ip/pinmux/data/pinmux_fpv_testplan.hjson:193]:
```suggestion
      desc: '''If not at posedge of `sleep_en_i`, and in the meantime register
```
And below

File: hw/ip/pinmux/data/pinmux_fpv_testplan.hjson [hunk 3]
```diff
@@ -5,121 +5,795 @@
...
+    }
+    {
+      name: DftStrapTestOStrapStable_A
+      desc: "`dft_strap_test_o.valid` once set, `dft_strap_test_o.straps` should stay stable."
+      milestone: V1
+      tests: ["pinmux_assert"]
+    }
+
+    // USB related IOs.
+    // Current plan is to only check connectivity because usbdev design is fully verified
```
Comment (matutem) [inline hw/ip/pinmux/data/pinmux_fpv_testplan.hjson:764]: Are these connectivity checks in pinmux_assert, or in the connectivity CSVs?

File: hw/ip/pinmux/data/pinmux_fpv_testplan.hjson [hunk 4]
```diff
@@ -5,121 +5,795 @@
...
+      milestone: V2
+      tests: ["pinmux_assert"]
+    }
+
+    {
+      name: TapStrap_A
+      desc: '''If `dft_hold_tap_sel_i` is 0 and `lc_dft_en_i` is On (tn the past two clock cycles
+            due to the synchronizer), or `strap_en_i` is 1.
+            And in the meantime, if `lc_hw_debug_en_i` is On (at the past two clock cycles due to
+            the synchronizer), then tap_strap must be equal to the past value of corresponding
```
Comment (matutem) [inline hw/ip/pinmux/data/pinmux_fpv_testplan.hjson:700]:
```suggestion
      desc: '''If `dft_hold_tap_sel_i` is 0 and `lc_dft_en_i` is On for the past two clock cycles
            due to the synchronizer, or `strap_en_i` is 1.
            And in the meantime, if `lc_hw_debug_en_i` is On for the past two clock cycles due to
            the synchronizer, then tap_strap must be equal to the past value of corresponding
```
And below.

File: hw/ip/pinmux/data/pinmux_fpv_testplan.hjson [hunk 5]
```diff
@@ -5,121 +5,795 @@
...
+            - Lc Jtag is enabled and the corresponding pins are 1.
+            '''
+      milestone: V2
+      tests: ["pinmux_assert"]
+    }
+
+    // `rv_jtag_o` tests
+    {
+      name: RvJtagWoScanmode_A
+      desc: '''Not in scanmode, when tap_strap select RV_tap and `lc_hw_debug_en_i` input is On (at
```
Comment (matutem) [inline hw/ip/pinmux/data/pinmux_fpv_testplan.hjson:611]: Here and below, use "for the past two ...", and no need to parenthesize it.

**Review by matutem — APPROVED** [review]
A few suggested changes, and then LGTM.

File: hw/ip/pinmux/data/pinmux_fpv_testplan.hjson [hunk 6]
```diff
@@ -5,121 +5,795 @@
...
+            corresponding `mio_oe_o` must be 1.'''
+      milestone: V1
       tests: ["pinmux_assert"]
     }
     {
       name: OutSelOe2_A
-      desc: "When register mio_outsel is set to 2, the corresponding mio_out_o must be 0."
-      milestone: V2
+      desc: '''When register `mio_outsel` is set to 2 and is not in sleep mode or jtag, which
+            indicates driving high-Z to the selected output, so the corresponding `mio_oe_o` must
```
[the remark this answers is not in the document]
Comment (cindychip) [reply hw/ip/pinmux/data/pinmux_fpv_testplan.hjson]: thanks again! Updated

File: hw/ip/pinmux/data/pinmux_fpv_testplan.hjson (hunk 3 above)
Comment (cindychip) [reply hw/ip/pinmux/data/pinmux_fpv_testplan.hjson]: It is just an assertion to check the connectivity, not the CSVs. Let me update the description to clear the ambiguity. Thanks for pointing that out.

**Pull request merged**